### PR TITLE
adding some static test in the "testing mode"

### PR DESCRIPTION
**Pull request by fanff**
For simiplified testing without backed

### CODE REVIEW

**Review by Zimzozaur — COMMENTED** [review]
Shouldn't we create a folder for testing?

**Comment by fanff** [comment]
@Zimzozaur  updated the data in a data folder with static data only. 

The api got a client; (actually two). 

* The client keeps the api request.get /post/ and formating  in its class. 
* The essential method that can be mocked up are done in the PulporoPyStaticClient ; that can run without any server for testing purposes. 

This update can allow futher upgrade later with mockup data and upgraded headers with login/password etc..

File: apiclient.py
Comment (Zimzozaur) [inline apiclient.py:1]: I would call it api_client for sake of readability as all other files

File: tests/data.py
Comment (Zimzozaur) [inline tests/data.py:1]: The data in the dict does not mock the DB model. Could we create a script in the util folder that creates a fixture and then just return it?

File: views/ledger.py [hunk 1]
```diff
@@ -110,7 +112,8 @@ class Ledger(Container):
         'year': TODAY.year,
         'month': TODAY.month,
     }
-
+    client = PulporoPyClient(PULPORO_URL) if PULPORO_URL!="TEST" else PulporoPyStaticClient()
```
Comment (Zimzozaur) [inline views/ledger.py:115]: Is it a good practice to write this type of condition in production code? Isn't it better to just write a test case where you dynamically assign a new value to a client attribute?

File: views/ledger.py [hunk 2]
```diff
@@ -180,12 +183,7 @@ def swap_table_to_new_update_month_button(boolean):
 
     def request_table_data(self) -> list[tuple]:
         """Call Pulporo endpoint and return list of tuples"""
-
-        if PULPORO_URL=="TEST":
-            list_of_dicts = test_static_data(self.ENDPOINT_URL)
```
Comment (Zimzozaur) [inline views/ledger.py]: I would make it generic that it is not essential that it is a test end point or production

**Review by Zimzozaur — CHANGES_REQUESTED** [review]
I like the idea of api_client, but I am not sure about fake data loader

File: tests/data.py
Comment (fanff) [reply tests/data.py:1]: stash for later :)

File: views/ledger.py (hunk 1 above)
Comment (Zimzozaur) [reply views/ledger.py:115]: Okay, so done is better then perfect

**Pull request merged**